Network/DNS/StateBinary.hs

```diff
@@ -136,7 +136,7 @@ get32 = ST.lift getWord32be <* addPosition 4
         b <- word8'
         c <- word8'
         d <- word8'
-        return $ a * 1677721 + b * 65536 + c * 256 + d
+        return $ a * 0x1000000 + b * 0x10000 + c * 0x100 + d
```

Comment:
Thank you for providing this library. *Soooo* much easier to extend this than starting from scratch.